src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -219,6 +219,12 @@ public function buildView(FormView $view, FormInterface $form, array $options)
             // POST request.
             $view->vars['full_name'] .= '[]';
         }
+
+        // Provide a default message when choices var is empty
+        if (empty($view->vars['choices']) && $options['expanded']) {
```

Comment:
I would remove this `if` completely. Its up to the template to decided what to display and when.
At least I would remove the check for `expanded` since one could decide to display a message insteadof an empty select box.